### DISCUSSION TITLE
HammerDB Error : Error in Virtual User 1: ERROR: syntax error at or near "@"

### DISCUSSION THREAD

**Discussion by sreejith-g-github** [TPC-Council/HammerDB#146] [Q&A] [answered]
Hi Team,

I am a first time user of HammerDB. Trying to run benchmark for Postgres Enterprise DB using HammerDB in Windows server 2019.   I am using the superuser credentials ( postgres ) user.
Odbc test connection has been succesful . Hammerdb is throwing error **Error in Virtual User 1: ERROR: syntax error at or near "@"** while trying to build schema. Any help to resolve this is appreciated.

Regards,
Sree

**Comment by sm-shaw** [comment]
Its difficult to know where you are going wrong without more details.  Note that access is configured in the file pg_hba.conf in the data directory you will need to update this especially if accessing over the network. 
Here is a PostgreSQL on windows example:
1. Start the database
```
E:\PgSQL\binaries\pgsql\bin>pg_ctl start -D E:\PgSQL\binaries\pgsql\data
waiting for server to start....2020-09-15 10:12:42.275 BST [8664] LOG:  starting PostgreSQL 12.0, compiled by Visual C++ build 1914, 64-bit
2020-09-15 10:12:42.280 BST [8664] LOG:  listening on IPv6 address "::1", port 5432
2020-09-15 10:12:42.280 BST [8664] LOG:  listening on IPv4 address "127.0.0.1", port 5432
2020-09-15 10:12:42.411 BST [760] LOG:  database system was shut down at 2020-09-15 10:12:38 BST
2020-09-15 10:12:42.482 BST [8664] LOG:  database system is ready to accept connections
 done
server started
```
Configure the build parameters and build the schema:
![pgbuild](https://user-images.githubusercontent.com/38044085/93194704-4b35b500-f740-11ea-876b-fd35a5a621a4.PNG)

Check that it built OK and we can login manually with the user we created. 
```
E:\PgSQL\binaries\pgsql\bin>psql -d postgres
psql (12.0)
WARNING: Console code page (850) differs from Windows code page (1252)
         8-bit characters might not work correctly. See psql reference
         page "Notes for Windows users" for details.
Type "help" for help.

postgres=# \l
                                              List of databases
   Name    | Owner | Encoding |           Collate           |            Ctype            | Access privileges
-----------+-------+----------+-----------------------------+-----------------------------+-------------------
 postgres  | Steve | WIN1252  | English_United Kingdom.1252 | English_United Kingdom.1252 |
 template0 | Steve | WIN1252  | English_United Kingdom.1252 | English_United Kingdom.1252 | =c/Steve         +
           |       |          |                             |                             | Steve=CTc/Steve
 template1 | Steve | WIN1252  | English_United Kingdom.1252 | English_United Kingdom.1252 | =c/Steve         +
           |       |          |                             |                             | Steve=CTc/Steve
 tpcc      | tpcc  | WIN1252  | English_United Kingdom.1252 | English_United Kingdom.1252 |
(4 rows)
postgres=#

E:\PgSQL\binaries\pgsql\bin>psql -U tpcc -d tpcc
psql (12.0)
WARNING: Console code page (850) differs from Windows code page (1252)
         8-bit characters might not work correctly. See psql reference
         page "Notes for Windows users" for details.
Type "help" for help.

tpcc=> \d
          List of relations
 Schema |    Name    | Type  | Owner
--------+------------+-------+-------
 public | customer   | table | tpcc
 public | district   | table | tpcc
 public | history    | table | tpcc
 public | item       | table | tpcc
 public | new_order  | table | tpcc
 public | order_line | table | tpcc
 public | orders     | table | tpcc
 public | stock      | table | tpcc
 public | warehouse  | table | tpcc
(9 rows)
```
Configure the driver to run as tpcc with password tpcc
![postgresqllogin](https://user-images.githubusercontent.com/38044085/93195062-b54e5a00-f740-11ea-8826-468bec947021.PNG)

Run the test

![pgtest](https://user-images.githubusercontent.com/38044085/93195731-7cfb4b80-f741-11ea-9c3f-59706dae8cd4.PNG)

Shutdown the database
```
E:\PgSQL\binaries\pgsql\bin>pg_ctl stop -D E:\PgSQL\binaries\pgsql\data
waiting for server to shut down............ done
server stopped
```

**Comment by sreejith-g-github** [comment]
Thanks Steve for helping out. Let me elaborate. Running Hammerdb tool v 3.2 and Postgres sql Enterprise db on windows 2019 server. 

I could connect to Postgres DB using PGadmin and also could create/ delete Databases.

While trying to build schema, virtual users get errors as attached the screenshots. Please let me know if I am doing anything wrong specifically
![error3](https://user-images.githubusercontent.com/66629814/93243434-cf6e5380-f7a5-11ea-98d3-e054332105d1.png)
![error2](https://user-images.githubusercontent.com/66629814/93243449-d5643480-f7a5-11ea-80a7-7ab9eba7ec2e.png)
![error1](https://user-images.githubusercontent.com/66629814/93243462-da28e880-f7a5-11ea-846b-0a0cdc5548d8.png)

**Comment by sm-shaw** [comment, marked as answer]
Try without the '@' symbol in usernames or passwords

**Comment by sreejith-g-github** [comment]
Thanks Steve. Could get this to work. There were lot of stale entries for roles,db's,passwords  due to various tries. Cleaned up everything and tried without @ symbol. It worked. Thank you for your time.
